thor/reader.py

```diff
@@ -104,72 +103,104 @@ def getArea(self,
             else:
                 return {"ok": False}
 
-        startTime = int(round((fromDate-self.startDate).days /
-                              self.dateResolution))
-        stopTime = int(round((toDate-self.startDate).days /
-                             self.dateResolution))
-        # Due to how numpy range-indexing works, we need one more.
-        if stopTime < len(self.netCDF.variables["time"]) - 1:
-            stopTime = stopTime + 1
+        startLong = 0
+        while self.netCDF.variables['rlon'][
+                startLong] < rotFrom.item(0, 0):
+            if startLong < self.lonLen:
+                startLong = startLong + 1
+            else:
+                return {"ok": False}
+
+        stopLong = startLong + 1
+        while self.netCDF.variables['rlon'][
+              stopLong] < rotTo.item(0, 0):
+            if stopLong < self.lonLen:
+                stopLong = stopLong + 1
+            else:
+                return {"ok": False}
 
         return({"ok": True,
-                "data": [startLong,
-                         stopLong,
+                "data": [startTime,
+                         stopTime,
                          startLat,
                          stopLat,
-                         startTime,
-                         stopTime]})
-
+                         startLong,
+                         stopLong]})
+
+    # -------------------------------------
+    def interpolate(self,
+                    values,
+                    maxTime,
+                    maxLat,
+                    maxLong,
+                    returnDimension):
+
+        timeCoord1D = np.linspace(0, maxTime-1, maxTime)
+        latCoord1D = np.linspace(0, maxLat-1, maxLat)
+        longCoord1D = np.linspace(0, maxLong-1, maxLong)
+
+        points = (timeCoord1D,
+                  latCoord1D,
+                  longCoord1D)
+
+        weatherInterpolationFunc = scipy.interpolate.RegularGridInterpolator(
+            points,
+            values)
+
+        returnData3D = np.ndarray(returnDimension,
+                                  dtype=float)
+
+        i = 0
+        j = 0
+        k = 0
+        for time in np.linspace(0, maxTime-1, returnDimension[0]):
+            for lat in np.linspace(0, maxLat-1, returnDimension[1]):
+                for long in np.linspace(0, maxLong-1, returnDimension[2]):
+                        returnData3D[i, j, k] = weatherInterpolationFunc((
+                            time,
+                            lat,
+                            long))
```

Comment:
It might be possible to get it to work faster by using https://docs.scipy.org/doc/scipy-0.18.1/reference/generated/scipy.interpolate.griddata.html

Also @Rovanion talked about using the map function to get around having to use the for loops.